sw/device/tests/rv_core_ibex_nmi_irq_test.c

```diff
@@ -0,0 +1,271 @@
+  top_earlgrey_plic_peripheral_t peripheral;
+  dif_alert_handler_irq_t irq;
+  isr_testutils_alert_handler_isr(plic_ctx, alert_handler_ctx, &peripheral,
+                                  &irq);
+
+  // Check that the alert handler escalation can be stoped.
+  bool can_clear;
+  CHECK_DIF_OK(dif_alert_handler_escalation_can_clear(
+      alert_handler_ctx.alert_handler, kDifAlertHandlerClassA, &can_clear));
+  CHECK(can_clear, "Alert handler is locked.");
+
+  CHECK_DIF_OK(dif_alert_handler_escalation_clear(
+      alert_handler_ctx.alert_handler, kDifAlertHandlerClassA));
+}
+
+/**
+ * Execute the wdog bark nmi interrupt test.
+ */
+static void wdog_nmi_test(void) {
+  rv_plic_testutils_irq_range_enable(
+      &plic, plic_ctx.hart_id, kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark,
+      kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark);
+
+  CHECK_DIF_OK(
+      dif_rv_core_ibex_enable_nmi(&rv_core_ibex, kDifRvCoreIbexNmiSourceWdog));
+
+  // Setup to handle the incoming wdog nmi irq.
+  expected_isr_handler = wdog_handler;
+  nmi_fired = false;
+  nmi_state = (dif_rv_core_ibex_nmi_state_t){0};
+
+  // Setup the wdog bark interrupt.
+  uint32_t count_cycles =
+      aon_timer_testutils_get_aon_cycles_from_us(kWdogBarkMicros);
+  aon_timer_testutils_watchdog_config(&aon_timer,
+                                      /*bark_cycles=*/count_cycles,
+                                      /*bite_cycles=*/UINT32_MAX,
+                                      /*pause_in_sleep=*/false);
+  IBEX_SPIN_FOR(nmi_fired, kWdogBarkMicros * 2);
+
+  // Check that the wdog nmi was enabled and barked during the nmi irq.
+  CHECK(nmi_state.wdog_enabled && nmi_state.wdog_barked,
+        "Wdog nmi state not expected:\n\t"
+        "wdog_enable:%x\n\twdog_raised:%"
+        "x\n",
+        nmi_state.wdog_enabled, nmi_state.wdog_barked);
+
+  // Check that the wdog nmi is enable and is cleared.
+  CHECK_DIF_OK(dif_rv_core_ibex_get_nmi_state(
+      &rv_core_ibex, (dif_rv_core_ibex_nmi_state_t *)&nmi_state));
+  CHECK(nmi_state.wdog_enabled && !nmi_state.wdog_barked,
+        "Wdog nmi state not expected:\n\t"
+        "wdog_enable:%x\n\twdog_raised:%"
+        "x\n",
+        nmi_state.wdog_enabled, nmi_state.wdog_barked);
+}
+
+/**
+ * Execute the alert handler nmi interrupt test.
+ */
+static void alert_handler_nmi_test(void) {
+  alert_handler_config();
+  rv_plic_testutils_irq_range_enable(&plic, plic_ctx.hart_id,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa);
+  CHECK_DIF_OK(
+      dif_rv_core_ibex_enable_nmi(&rv_core_ibex, kDifRvCoreIbexNmiSourceAlert));
+
+  // Setup to handle the incoming alert handler nmi irq.
+  expected_isr_handler = alert_handler_handler;
+  nmi_fired = false;
+
+  // Trigger the alert handler to escalate.
+  CHECK_DIF_OK(dif_pwrmgr_alert_force(&pwrmgr, kDifPwrmgrAlertFatalFault));
+
+  IBEX_SPIN_FOR(nmi_fired, 10000);
```

Comment:
since the force latency / nmi should be pretty short (unlike the watchdog case), should we have a variable for this delay also?

Reply:
the nmi fires immediately, I change it to use the `kEscalationPhase0Micros` here.

Reply:
thanks @engdoreis